server/game/gamesteps/abilityresolver.js

```diff
@@ -231,7 +231,7 @@ class AbilityResolver extends BaseStep {
             this.game.resolveEvent(event);
         }
         if(this.context.pull) {
-            this.context.player.handlePulledCard(this.context.pull.pulledCard);
+            this.context.pull.pulledCard.owner.handlePulledCard(this.context.pull.pulledCard);
```

Comment:
Changed the player doing the pulled card handling to the owner of the card so it will go to proper discard pile. Same was done in the `player.handlePull` function.